test/Microsoft.ComponentDetection.Detectors.Tests/NpmDetectorWithRootsTests.cs

```diff
@@ -580,4 +606,71 @@ public async Task TestNpmDetector_DependencyGraphIsCreatedAsync()
         dependencyGraph.GetDependenciesForComponent(componentBId).Should().Contain(componentCId);
         dependencyGraph.GetDependenciesForComponent(componentCId).Should().HaveCount(0);
     }
+
+    [TestMethod]
+    public async Task TestNpmDetector_NestedNodeModulesV3Async()
```

Comment:
Is this test case representative of a real `package-lock.json` file? Wouldn't component B get hoisted to the top level?

You also need another test case to cover the case
```
root -> package A -> package B (v2)
     \-> package B (v1)
```

Reply:
Yep, this was created by simply adding the `debug` package:

```json
{
  "name": "test-pkg",
  "version": "1.0.0",
  "lockfileVersion": 3,
  "requires": true,
  "packages": {
    "": {
      "name": "test-pkg",
      "version": "1.0.0",
      "license": "ISC",
      "dependencies": {
        "debug": "^4.3.4"
      }
    },
    "node_modules/debug": {
      "version": "4.3.4",
      "resolved": "https://registry.npmjs.org/debug/-/debug-4.3.4.tgz",
      "integrity": "sha512-PRWFHuSU3eDtQJPvnNY7Jcket1j0t5OuOsFzPPzsekD52Zl8qUfFIPEiswXqIvHWGVHOgX+7G/vCNNhehwxfkQ==",
      "dependencies": {
        "ms": "2.1.2"
      },
      // ...
    },
    "node_modules/ms": {
      "version": "2.1.2",
      "resolved": "https://registry.npmjs.org/ms/-/ms-2.1.2.tgz",
      "integrity": "sha512-sGkPx+VjMtmA6MX27oA4FBFELFCZZ4S4XqeGOXCv68tT+jb3vk/RyaKWP0PTKyWtmLSM0b+adUTEvbs1PEaH2w=="
    }
  }
}
```

Reply:
Yes, this configuration can be created by this sequence:

```bash
npm install send
npm install debug
npm remove send
```

This gives you this `package-lock.json`:

```json
{
  "name": "test-pkg",
  "version": "1.0.0",
  "lockfileVersion": 3,
  "requires": true,
  "packages": {
    "": {
      "name": "test-pkg",
      "version": "1.0.0",
      "license": "ISC",
      "dependencies": {
        "debug": "^4.3.4"
      }
    },
    "node_modules/debug": {
      "version": "4.3.4",
      "resolved": "https://registry.npmjs.org/debug/-/debug-4.3.4.tgz",
      "integrity": "sha512-PRWFHuSU3eDtQJPvnNY7Jcket1j0t5OuOsFzPPzsekD52Zl8qUfFIPEiswXqIvHWGVHOgX+7G/vCNNhehwxfkQ==",
      "dependencies": {
        "ms": "2.1.2"
      },
      // ...
    },
    "node_modules/debug/node_modules/ms": {
      "version": "2.1.2",
      "resolved": "https://registry.npmjs.org/ms/-/ms-2.1.2.tgz",
      "integrity": "sha512-sGkPx+VjMtmA6MX27oA4FBFELFCZZ4S4XqeGOXCv68tT+jb3vk/RyaKWP0PTKyWtmLSM0b+adUTEvbs1PEaH2w=="
    }
  }
}
```